modules/search/instant-search/components/search-app.jsx

```diff
@@ -61,56 +60,91 @@ class SearchApp extends Component {
 		} );
 	}
 
+	componentDidMount() {
+		if ( this.props.grabFocus ) {
+			this.input.current.focus();
+		}
+	}
+
+	hasActiveQuery() {
+		return this.state.query !== '' || hasFilter();
+	}
+
+	activateResults() {
+		if ( ! this.state.resultsActive ) {
+			this.setState( { resultsActive: true } );
+			hideChildren( this.props.themeOptions.results_selector );
+		}
+	}
+
+	maybeDeactivateResults() {
+		if ( this.isSearchPage || this.hasActiveQuery() ) {
+			return;
+		}
+		if ( this.state.resultsActive ) {
+			this.setState( { resultsActive: false } );
```

Comment:
Ditto here about using [a callback for `setState`](https://reactjs.org/docs/react-component.html#setstate):
`this.setState( { resultsActive: false }, () => { doSomethingHere(); } );`